packages/@aws-cdk/aws-ecs-patterns/lib/base/scheduled-task-base.ts

```diff
@@ -152,22 +152,12 @@ export abstract class ScheduledTaskBase extends CoreConstruct {
   }
 
   /**
-   * Create an ECS task using the task definition provided and add it to the scheduled event rule.
+   * Adds task as a target of the scheduled event rule.
    *
-   * @param taskDefinition the TaskDefinition to add to the event rule
+   * @param ecsTaskTarget the EcsTask to add to the event rule
    */
-  protected addTaskDefinitionToEventTarget(taskDefinition: TaskDefinition): EcsTask {
-    // Use the EcsTask as the target of the EventRule
-    const eventRuleTarget = new EcsTask( {
-      cluster: this.cluster,
-      taskDefinition,
-      taskCount: this.desiredTaskCount,
-      subnetSelection: this.subnetSelection,
-    });
-
-    this.eventRule.addTarget(eventRuleTarget);
-
-    return eventRuleTarget;
+  protected addTaskAsTarget(ecsTaskTarget: EcsTask) {
+    this.eventRule.addTarget(ecsTaskTarget);
```

Comment:
Technically this would be a backwards breaking change. Can you keep the existing method, and have it call your new method ... something like below:

```
protected addTaskDefinitionToEventTarget(taskDefinition: TaskDefinition): EcsTask {
    // Use the EcsTask as the target of the EventRule
    const eventRuleTarget = new EcsTask( {
      cluster: this.cluster,
      taskDefinition,
      taskCount: this.desiredTaskCount,
      subnetSelection: this.subnetSelection,
    });

    addTaskAsTarget(eventRuleTarget);

    return eventRuleTarget;
}

protected addTaskAsTarget(ecsTaskTarget: EcsTask) {
    this.eventRule.addTarget(ecsTaskTarget);
}
```

Reply:
Yeah sure, I'll push up this change later today :)

Reply:
Awesome! Thanks a lot for adding this feature @richardhughes!